### PR TITLE
fix closure imports by appending /index where needed

### PR DESCRIPTION
**Pull request by alexeagle**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by alexeagle** [comment]
Hmm, this is currently blocked.

The issue is that tsickle now rewrites `import {} from '@angular/core'` to `import {} from '@angular/core/index'`. This is in the ES5 JS sources. Then when we bundle these for UMD, we get eg. compiler/bundles/compiler.umd.js containing:
```
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core/index')) :
    typeof define === 'function' && define.amd ? define(['exports', '@angular/core/index'], factory) :
    (factory((global.ng = global.ng || {}, global.ng.compiler = global.ng.compiler || {}),global._angular_core_index));
}(this, function (exports,_angular_core_index) { 'use strict';
```

here, the deep import from `@angular/core/index` causes the loader to skip the UMD bundle for core, and instead load ES2015 module syntax from index.js.

**Comment by alexeagle** [comment]
This is now unblocked, @IgorMinar you should probably take a look since this adds the experimental ES6 build (guarded by an env variable)

**Comment by IgorMinar** [comment]
We should update the Angular Package Format doc before merging these changes to truly understand the impact. I read through the PR but I'll need to checkout the branch and actually build it to see what are all the changes.

Let's not merge this until the design doc is updated. I'll work with Alex on this once he's back from vacation. I believe that this is PR is not super urgent and can wait a few days. If not please let me know.

**Comment by alexeagle** [comment]
This PR shouldn't change the existing build, because the features are guarded by a flag.
It's okay to wait until second week of January. Just some rebasing sadness, hopefully nothing big landing in the meantime...

**Comment by IgorMinar** [comment]
please fix commit messages:
- chore(tsc-wrapped) -> build(tsc-wrapped)
- feature(tsc-wrapped) -> feat(tsc-warpped) x 2
- feat(build): build: ....

File: tools/@angular/tsc-wrapped/src/compiler_host.ts [hunk 1]
```diff
@@ -99,9 +99,10 @@ export class TsickleCompilerHost extends DelegatingHost {
         // Don't tsickle-process any d.ts that isn't a compilation target;
         // this means we don't process e.g. lib.d.ts.
         if (isDefinitions) return sourceFile;
-
-        let {output, externs, diagnostics} =
-            tsickle.annotate(this.oldProgram, sourceFile, {untyped: true});
+        const es2015Target = this.options.target == ts.ScriptTarget.ES2015;  // This covers ES6 too
```
Comment (IgorMinar) [inline tools/@angular/tsc-wrapped/src/compiler_host.ts]: covers? it's equivalent. no?

File: build.sh [hunk 2]
```diff
@@ -131,6 +132,13 @@ do
 
   echo "======      COMPILING: ${TSC} -p ${SRCDIR}/tsconfig-build.json        ====="
   $TSC -p ${SRCDIR}/tsconfig-build.json
+  # ES2015 distro is not ready yet; don't slow down all builds for it
+  # TODO(alexeagle,igorminar): figure out ES2015 story and enable
```
Comment (IgorMinar) [inline build.sh]: discussed in person

**Review by IgorMinar — CHANGES_REQUESTED** [review]

File: tools/@angular/tsc-wrapped/src/compiler_host.ts (hunk 1 above)
Comment (alexeagle) [inline tools/@angular/tsc-wrapped/src/compiler_host.ts]: meaning this conditional matches if you pass `--target=ES6` on the command line.
ES6 and ES2015 get the same integer value

**Comment by alexeagle** [comment]
Done, PTAL

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**